src/main/scala/viper/server/ViperCoreServer.scala

```diff
@@ -0,0 +1,445 @@
+      VerificationJobHandler(-1) // Not able to create a new JobHandle
+    }
+  }
+
+  def stop(): Unit = {
+    println(s"Stopping ViperCoreServer")
+
+    getInterruptFutureList() onComplete {
+      case Success(_) =>
+        _term_actor ! Terminator.Exit
+        println(s"shutting down...")
+      case Failure(err_msg) =>
+        println(s"Interrupting one of the verification threads timed out: $err_msg")
+        _term_actor ! Terminator.Exit
+        println(s"forcibly shutting down...")
+    }
+  }
+
+
+  // --- VCS : Auxiliary Functions ---
+
+  protected def getInterruptFutureList(): Future[List[String]] = {
+    val interrupt_future_list: List[Future[String]] = _job_handles map { case (jid, handle_future) =>
+      handle_future.flatMap {
+        case JobHandle(actor, _, _) =>
+          implicit val askTimeout: Timeout = Timeout(5000 milliseconds)
+          (actor ? Stop(true)).mapTo[String]
+      }
+    } toList
+    val overall_interrupt_future: Future[List[String]] = Future.sequence(interrupt_future_list)
+    overall_interrupt_future
+  }
+
+  def flushCache(): Unit = {
+    ViperCache.resetCache()
+    println(s"The cache has been flushed successfully.")
+  }
+
+  /**
+   * Returns the future of the Verification Result.
+   * Deletes the jobhandle on completion.
+   */
+  def getFuture(jid: Int): Future[VerificationResult] = {
+    lookupJob(jid) match {
+      case Some(handle_future) =>
+        val result_future = handle_future.flatMap(handle => {
+          val sink = Sink.fold[Seq[AbstractError], Message](Seq())((errors, msg) => msg match {
+            case EntityFailureMessage(_, _, _, VerificationFailure(errs)) => errs ++ errors
+            case _ => errors
+          })
+
+          val errors_future: Future[Seq[AbstractError]] = Source.fromPublisher(handle.publisher)
+                                                                .toMat(sink)(Keep.right)
+                                                                .run()
+
+          val errors = errors_future.map({
+            case Seq() => VerificationSuccess
+            case errs => VerificationFailure(errs)
+          })
+
+          _term_actor ! Terminator.WatchJobQueue(jid, handle)
+          errors
+        })
+        result_future
+      case None =>
+        val promise = Promise[VerificationResult]()
+        promise failure (new NoSuchElementException(s"The verification job #$jid does not exist."))
+        promise.future
+    }
+  }
+
+//  private var test_actor: ActorRef = _
+
+  object PublishActor {
+    def props(reporter: Reporter): Props = Props(new PublishActor(reporter))
+  }
+
+  class PublishActor(reporter: Reporter) extends Actor {
+
+    override def receive: PartialFunction[Any, Unit] = {
+      case viper_msg: Message =>
+//        println(s"This is a Viper message ===============")
+//        println(viper_msg)
+        reporter.report(viper_msg)
```

Comment:
This seems to be the short-circuiting again. As soon as the publisher gets a new message, it instantly reports in via the reporter. I think we need something like a streaming reporter that would be connected to the source queue. 

To get an idea about how the source queue is supposed to help, I suggest taking a look at the Akka docs: https://doc.akka.io/docs/akka/current/stream/operators/Source/queue.html

They show an example in which throttle is used to artificially slow down the production of messages. 

Another scenario is when the client is slow: this could be modelled by adding ```sleep(1)``` (or whatever) before consuming each message. 

Now, with these two ways of controlling the rate of production/consumption, could you please add tests that can demonstrate what would happen in the following cases? 

1. The server produces verification results faster than the client can process them. The messages get buffered in the queue. The buffer grows until it hits the maximum size. Then the server reports an exception and closes the queue early. 

2. The server produces messages slower than the client consumes them. The buffer is populated with maximum 1 message at a time. 

It will be useful to make the maximum buffer size a command line argument (currently it's hardcoded as 10000). For testing, the buffer size should be set to something very small, like 5 or 10. 

It might also be useful in testing to create your dummy Verifier that produces a new verification result each second, or something like that.